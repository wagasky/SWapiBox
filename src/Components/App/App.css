.App {
  background-color: black;
  height: 100%;
  margin: 0;
  padding: 0;
  text-align: center;
  min-height: 100vh;
  width: 100%;
  /*height: 100%;*/
  background-color: black;
}

p { 
  color: white; 
}

.loading {
  background-color: black;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
}
