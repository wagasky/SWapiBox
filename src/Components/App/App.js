import React, { Component } from 'react';
import Header from '../Header/Header'
import CardDisplay from '../CardDisplay/CardDisplay'
import { BrowserRouter } from 'react-router-dom'
import PropTypes from 'prop-types'; 
import './App.css';
import { getPeople, getPlanets, getVehicles, getFilm } from '../../apiCalls/apiCalls'

class App extends Component {

  constructor(props) {
    super(props)

    this.state = {
      people: [],
      planets: [],
      vehicles: [],
      favorites: [],
      film: [],
    }
  }

  async componentDidMount() {
    let people = await getPeople(); 
    let planets = await getPlanets();
    let vehicles = await getVehicles();
    let film = await getFilm();
    this.setState({ people, planets, vehicles, film })
  }

  // planets and vehicle on click

  render() {
    return (
      <div className="App">
        <Header />
        <CardDisplay people={this.state.people} 
                     planets={this.state.planets}
                     vehicles={this.state.vehicles}
                     favorites={this.state.favorites}
                     film={this.state.film}
         />
      </div>
    );
  }
}

export default App;
