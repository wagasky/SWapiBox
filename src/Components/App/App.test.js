/* eslint-disable */ 

import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {

  let renderedComponent;

  beforeEach(() => {
   renderedComponent = shallow(<App />, {disableLifecycleMethods: true});
  })

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });
});


// test cDM
// test that state is set
// test that storage is set

// test the initial data calls

// handle button
// if it's given planets, it calls planets
// if it's given vehicles, etc.

// put into storage

// handlefavorite
// if a favorite doesn't exist, it gets put into storage
// if it does, it gets taken out of storage
// state is updated as well in both

// render - snapshot