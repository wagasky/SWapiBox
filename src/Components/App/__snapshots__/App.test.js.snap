// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App state should match snapshot 1`] = `
<div
  className="App"
>
  <Header
    handleButton={[Function]}
  />
  <CardDisplay
    favorites={Array []}
    film={Array []}
    handleFavorite={[Function]}
    people={Array []}
    planets={Array []}
    vehicles={Array []}
  />
</div>
`;
